provider/errors.go

```diff
@@ -49,3 +50,32 @@ func (err ErrUnknownProvider) Error() string {
 	}
 	return errStr.String()
 }
+
+// ErrInvalidProviderType is return when the requested provider type is not known for
+// the given name
+type ErrInvalidProviderType struct {
+	Name           string
+	Type           providerType
+	KnownProviders []string
+}
+
+func (err ErrInvalidProviderType) Error() string {
+	var errStr strings.Builder
+	fmt.Fprintf(&errStr, "provider '%v' is not of type %v", err.Name, err.Type)
+	if len(err.KnownProviders) != 0 {
+		fmt.Fprintf(&errStr, ", known providers of type (%v):", err.Type)
```

Comment:
Is this log intentionally left in?

Reply:
Nope

Reply:
Looking at this again. That is not a log statement, but building up the error string to return.
It's looks like a log statment, I know. I made that error as well.